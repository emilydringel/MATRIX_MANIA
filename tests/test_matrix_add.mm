def matrix<int> add(matrix<int> x, matrix<int> y){

    matrix<int> empty = [0,0];

    for(int i = 0; i < 2; i++){
    	for(int j = 0; j < 2; j++){
            empty[i,j] = x[i,j]+y[i,j];
        }
  	return empty;
    }
}


def int main(){
  
    matrix<int> m = [1,4];
    matrix<int> n = [2,3];

    matrix<int> added = add(m,n);

    printm(m);
    printm(n);
    printm(added);
    printm(add([6,8],[9,10]));
}
