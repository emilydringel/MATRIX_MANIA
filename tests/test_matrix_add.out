[1,4]
[2,3]
[3,7]
[15,18]